src/fragments/Markdown.react.js

```diff
@@ -21,15 +29,15 @@ export default class DashMarkdown extends Component {
     }
 
     highlightCode() {
-        if (!window.hljs) {
-            // skip highlighting if highlight.js isn't found
-            return;
-        }
         if (this.mdContainer) {
             const nodes = this.mdContainer.querySelectorAll('pre code');
 
-            for (let i = 0; i < nodes.length; i++) {
-                window.hljs.highlightBlock(nodes[i]);
+            if (MarkdownHighlighter.hljs) {
+                for (let i = 0; i < nodes.length; i++) {
+                    MarkdownHighlighter.hljs.highlightBlock(nodes[i]);
+                }
+            } else {
+                MarkdownHighlighter.loadhljs();
```

Comment:
I think the existing implementation does not take full advantage of the `react-markdown` ability to get its renderer overriden -- similarly to https://github.com/plotly/dash-core-components/pull/711/files#diff-5b56ed82805404e28302aef5147d04a2R129 it should be possible to hook up the highlight directly onto the `code` renderer and simplify this component / usage.

Reply:
Issue created: https://github.com/plotly/dash-core-components/issues/724, will not be addressed as part of this PR.